sklearn/pipeline.py

```diff
@@ -253,33 +254,44 @@ def set_metadata_request(self, props):
                            "level and not at the pipeline level.")
 
     def _check_fit_params(self, **fit_params):
-        required_props = self.get_metadata_request().fit
-        print("fit params steps:", list(fit_params.keys()))
-        _validate_required_props(required_props, fit_params)
-
         fit_params_steps = {name: {} for name, step in self.steps
                             if step is not None}
-        """
-        for pname, pval in fit_params.items():
-            if '__' not in pname:
-                raise ValueError(
-                    "Pipeline.fit does not accept the {} parameter. "
-                    "You can pass parameters to specific steps of your "
-                    "pipeline using the stepname__parameter format, e.g. "
-                    "`Pipeline.fit(X, y, logisticregression__sample_weight"
-                    "=sample_weight)`.".format(pname))
-            step, param = pname.split('__', 1)
-            fit_params_steps[step][param] = pval
-        """
+        # Remove old_behavior in 0.26
+        old_behavior = np.any(['__' in pname for pname in fit_params.keys()])
+        if old_behavior:
+            warnings.warn("It seems fit_params are using the deprecated "
+                          "way to route parameters using '__'. This behavior "
+                          "is deprecated in 0.24 and won't be accepted in "
+                          "0.26. Please use `set_request_metadata` to route "
+                          "parameters.", FutureWarning)
+            for pname, pval in fit_params.items():
+                if '__' not in pname:
+                    raise ValueError(
+                        "Pipeline.fit does not accept the {} parameter. "
+                        "You can pass parameters to specific steps of your "
```

Comment:
This doesn't seem appropriate anymore. Need a message like "any non-standard parameter passed to Pipeline.fit should be requested by its consumer"

Reply:
This would only be raised if the warning is also raised, which tells users to use the new API. So technically, until the deprecation is completed, users can pass parameters as they were before.

Reply:
Although when deprecating, we often don't suggest that people use the deprecated behaviour. Might be good to encourage them to adopt the new behaviour, but no big deal.